### PR TITLE
Improve APM for integration tests

### PR DESCRIPTION
**Pull request by jirikuncar**
⬆️

### CODE REVIEW

File: .azure-pipelines/all.yml [hunk 1]
```diff
@@ -43,13 +47,16 @@ jobs:
...
         DD_AGENT_HOST: datadog-agent
-        DD_TAGS: "build.requested_by:$(Build.RequestedForId),build.attempt:$(System.StageAttempt),pull_request.number:$(System.PullRequest.PullRequestNumber)"
+        DD_ENV: prod
         DD_HTTP_CLIENT_ERROR_STATUSES: "500-599"
+        DD_LOGS_INJECTION: "true"
+        DD_SERVICE: datadog-api-client-java
+        DD_TAGS: "team:integration-tools-and-libraries,build.requested_by:$(Build.RequestedForId),build.attempt:$(System.StageAttempt),pull_request.number:$(System.PullRequest.PullRequestNumber)"
         DD_TEST_CLIENT_API_KEY: $(datadogAPIKey)
         DD_TEST_CLIENT_APP_KEY: $(datadogAPPKey)
+        DD_TRACE_ANNOTATIONS: org.junit.Test
```
Comment (nmuesch) [inline .azure-pipelines/all.yml:58]: Oh cool, does this give us auto instrumentation of every Test?

File: pom.xml [hunk 2]
```diff
@@ -339,6 +339,13 @@
             <version>${dd-java-agent-version}</version>
             <scope>test</scope>
         </dependency>
+        <!--
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-jdk14</artifactId>
+            <version>1.7.25</version>
+        </dependency>
+        -->
```
Comment (nmuesch) [inline pom.xml:348]: Do we still need this ^?

File: .azure-pipelines/all.yml (hunk 1 above)
Comment (jirikuncar) [reply .azure-pipelines/all.yml:58]: Exactly without custom annotations.

File: pom.xml (hunk 2 above)
Comment (jirikuncar) [reply pom.xml:348]: I am hesitant to enable it for tests only.

Comment (jirikuncar) [inline pom.xml:348]:
```suggestion
```

File: .azure-pipelines/all.yml [hunk 3]
```diff
@@ -43,13 +47,16 @@ jobs:
...
+        DD_TRACE_HEADER_TAGS: "Dd-Operation-Id:dd.operation.id"
```
Comment (jirikuncar) [inline .azure-pipelines/all.yml:59]:
```suggestion
```

File: docker-compose.yml [hunk 4]
```diff
@@ -11,11 +11,14 @@ services:
...
-      DD_TRACE_AGENT_PORT: 8126
-      DD_SERVICE_NAME: datadog-api-client-java
       DD_HTTP_CLIENT_ERROR_STATUSES: "500-599"
+      DD_LOGS_INJECTION: "true"
+      DD_SERVICE: datadog-api-client-java
       DD_TEST_CLIENT_API_KEY: ${DD_TEST_CLIENT_API_KEY}
       DD_TEST_CLIENT_APP_KEY: ${DD_TEST_CLIENT_APP_KEY}
+      DD_TRACE_AGENT_PORT: 8126
+      DD_TRACE_ANNOTATIONS: org.junit.Test
+      DD_TRACE_HEADER_TAGS: "Dd-Operation-Id:dd.operation.id"
```
Comment (jirikuncar) [inline docker-compose.yml:21]:
```suggestion
```

**Pull request merged**